src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        for (User registeredUser : Storage.people) {
+            if (registeredUser.getLogin().equals(user.getLogin())) {
+                throw new RegistrationException(
+                        "User with login " + user.getLogin() + " already exist");
+            }
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("User login should be at least 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException("Password should be at least 6 characters");
+        }
```

Comment:
The null check for `user.getPassword()` should be performed before any operations on the password, such as checking its length, to prevent potential `NullPointerException`.